pan3d/dataset_builder.py

```diff
@@ -261,77 +281,66 @@ def on_set_array_active(self, array_active, **kwargs):
                     "step": 1,
                 }
             )
-            self.state.expanded_coordinates.append(key),
+            self.state.ui_expanded_coordinates.append(key),
         self.auto_select_coordinates()
-        self.state.dirty("coordinates", "expanded_coordinates")
-
-    @change("x_array")
-    def on_set_x_array(self, x_array, **kwargs):
-        self.algorithm.x = x_array
-        self.mesh_changed()
-
-    @change("y_array")
-    def on_set_y_array(self, y_array, **kwargs):
-        self.algorithm.y = y_array
-        self.mesh_changed()
+        self.state.dirty("da_coordinates", "ui_expanded_coordinates")
+        self.plotter.clear()
+        self.plotter.view_isometric()
 
-    @change("z_array")
-    def on_set_z_array(self, z_array, **kwargs):
-        self.algorithm.z = z_array
+    @change("da_x", "da_y", "da_z")
+    def on_set_da_axis(self, da_x, da_y, da_z, **kwargs):
+        self.algorithm.x = da_x
+        self.algorithm.y = da_y
+        self.algorithm.z = da_z
         self.mesh_changed()
 
-    @change("t_array")
-    def on_set_t_array(self, t_array, **kwargs):
-        self.algorithm.time = t_array
-        if self.dataset and self.state.array_active and t_array:
-            time_steps = self.dataset[self.state.array_active][t_array]
+    @change("da_t")
+    def on_set_da_t(self, da_t, **kwargs):
```

Comment:
update_mesh_time?

Reply:
what is da_t vs da_t_index? I can't figure out what each method is doing?

Reply:
The intention of `da_t` is similar to `da_x`, etc. These variables store strings to refer to the name of the array. `da_t` stores the name of the time variable, which would typically be "time", but it could be something like "second" or "day". Whereas `da_t_index` stores an integer which is the current slice along that named axis.

Reply:
Thanks for the clarification